### DISCUSSION TITLE
Clickable Particle

### DISCUSSION THREAD

**Discussion by laffan** [matteobruni/tsparticles#1610] [Q&A] [answered]
I'm wondering if there is any support for `onClick` at the individual particle level? (I'm working with `react-tsparticles`.)

I'd like to pass each particle a little bit of data (maybe by sending an object array in to `manualParticles` ?) that can be attached/retrieved with a click. I'm not sure if that's possible, but it seems like it would be a beautiful/interesting way to explore a data set that doesn't need to be ordered.

Thank you in advance!

**Comment by matteobruni** [comment, marked as answer]
It's possible but I noticed that on React and other components it's a little tricky. I'm preparing 1.28.0 for release and I improve that in that version.

**Reply by laffan** [reply]
Good to know. Thanks for the quick response!

**Reply by matteobruni** [reply]
`1.28.0` is out now.

To retrieve clicked particles you have to set the property `loaded`, it's a callback with a `Container` object parameter. That object has now a `addClickHandler` method with a callback parameter.

The callback will be called with two parameters: event and particles, the first is the original event object, the second one is an array containing all the clicked particles.
